### DISCUSSION TITLE
When I use tp.date.now as part of a link, it doesn't work.

### DISCUSSION THREAD

**Discussion by dandobos** [SilentVoid13/Templater#842] [Help]
When I use:
[[<% tp.date.now("YYYY-MM-DD", -7, tp.file.title, "YYYY-MM-DD") %>]]
it works perfectly

But when I put it in a link it doesn't work :(
That is:
[One Week Ago]([[<% tp.date.now("YYYY-MM-DD", -7, tp.file.title, "YYYY-MM-DD") %>]])

Is there a way to modify the code so that the link works?

**Comment by AB1908** [comment]
Can you give us more than "it doesn't work"? Are you able to reliably reproduce this in the sandbox vault?

**Reply by dandobos** [reply]
Sorry I'm a beginner so I don't know what the sandbox vault is. The point is that the code that doesn't work means when you click the link, it doesn't go to the appropriate page.

**Reply by shabegom** [reply]
The link format you posted is wrong.

[[this is a wiki link]]

(This is a markdown link)[link]

You've put a wiki link inside a markdown link...

**Reply by AB1908** [reply]
Oh so that's what they meant by "it didn't work". My eyes just glazed right over that lol.

**Reply by dandobos** [reply]
> The link format you posted is wrong.
> 
> [[this is a wiki link]]
> 
> (This is a markdown link)[link]
> 
> You've put a wiki link inside a markdown link...

It was my intention to put a wikilink in a markdown link... because I want the markdown link to be a wikilink.... and the wikilink is created by templator... not sure if this possible?

**Reply by shabegom** [reply]
Uh what?

The only thing templater is doing here is generating a date:

`tp.date.now("YYYY-MM-DD", -7, tp.file.title, "YYYY-MM-DD")` returns `2022-09-19`

Because you are wrapping that date in square brackets it turns it into a wikilink: `[[2022-09-19]]`

The equivalent markdown link would be `[2022-09-19](2022-09-19)` assuming the note is at the root of your vault. If it is in a folder it would be `[2022-09-19](folder/2022-09-19)`

Anyways, your markdown link should look like:

`[One Week Ago](<% tp.date.now("YYYY-MM-DD", -7, tp.file.title, "YYYY-MM-DD") %>)`

Or you could just do:

`[[<% tp.date.now("YYYY-MM-DD", -7, tp.file.title, "YYYY-MM-DD") %>|One Week Ago]]`

**Reply by dandobos** [reply]
Thank you @shabegom - appreciate your help! :)
